### DISCUSSION TITLE
Case-sensitive feature tag?

### DISCUSSION THREAD

**Discussion by theraot** [godotengine/godot-proposals#7758] [Platforms]
I've been looking for a way to check if the game is running in a case-sensitive OS. Right now I can check against Windows, I also read that recent versions of macOS are not case sensitive, but I do not know which nor how how to check. Yet, if in the future more platforms are supported (should I worry about console support from W4Games?), I suppose I'd need to consider each scenario. I'd rather have a central way to check. Thus, does it make sense to add case-sensitive feature tag to Godot, or what would a better way to handle this?

**Comment by Calinou** [comment]
Does this actually make sense to be a feature tag, or should it be an OS singleton method such as `OS.is_filesystem_case_sensitive()`?

Remember that a disk may have multiple partitions with multiple case-sensitive statuses, although it's very unlikely that several partitions for the same OS are configured differently on this aspect.

Windows can also make certain folders case-sensitive, as documented in [Project organization](https://docs.godotengine.org/en/latest/tutorials/best_practices/project_organization.html#case-sensitivity).

**Reply by theraot** [reply]
That mostly what I was asking: does it make sense as a feature tag or something else? - Yes, I would like a method on the `OS` class, adding it to `DirAccess` did also cross my mind.

What didn't cross my mind is that it might differ from partition to partition. Now I have less of an idea of how to check.

**Comment by theraot** [comment]
With https://github.com/godotengine/godot/pull/82957 merged, I think this discussion doesn't serve any purpose anymore. Closing.
